hoomd/hpmc/pytest/test_convex_polygon.py

```diff
@@ -1,25 +1,132 @@
+
+
+def test_convex_poly_after_attaching(device, dummy_simulation_factory):
+
+    verts = [(-1, 1), (1, -1), (1, 1), (-1, -1)]
+    verts2 = [(-1, 1), (1, -1), (1, 1)]
+    poly = hpmc.integrate.ConvexPolygon(23456)
+    poly.shape['A'] = dict(vertices=verts)
+    poly.shape['B'] = dict(vertices=verts2, ignore_statistics=True)
+
+    sim = dummy_simulation_factory(particle_types=['A', 'B'])
+    sim.operations.add(poly)
+    sim.operations.schedule()
+
+    assert not poly.shape['A']['ignore_statistics']
+    assert poly.shape['B']['ignore_statistics']
+    np.testing.assert_allclose(poly.shape['A']['vertices'], verts)
+    np.testing.assert_allclose(poly.shape['B']['vertices'], verts2)
+
+
+def test_overlaps(device, dummy_simulation_check_overlaps):
+    mc = hoomd.hpmc.integrate.ConvexPolygon(23456, d=0, a=0)
+    mc.shape['A'] = dict(vertices=[(0, (0.75**0.5) / 2),
+                                   (-0.5, -(0.75**0.5) / 2),
+                                   (0.5, -(0.75**0.5) / 2)])
+    sim = dummy_simulation_check_overlaps()
+    sim.operations.add(mc)
+    # gsd_dumper = hoomd.dump.GSD(filename='/Users/danevans/hoomd/
+    # test_dump_convex_polygon.gsd', trigger=1, overwrite=True)
+    # gsd_logger = hoomd.logger.Logger()
+    # gsd_logger += mc
+    # gsd_dumper.log = gsd_logger
+    # sim.operations.add(gsd_dumper)
+    sim.operations.schedule()
+    sim.run(1)
+    overlaps = sim.operations.integrator.overlaps
+    assert overlaps > 0
+
+    s = sim.state.snapshot
+    s.particles.position[0] = (0, 0, 0)
+    s.particles.position[1] = (0, 8, 0)
+    sim.state.snapshot = s
+    sim.operations.add(mc)
+    # gsd_dumper = hoomd.dump.GSD(filename='/Users/danevans/hoomd/
+    # test_dump_convex_polygon.gsd', trigger=1, overwrite=True)
+    # gsd_logger = hoomd.logger.Logger()
+    # gsd_logger += mc
+    # gsd_dumper.log = gsd_logger
+    # sim.operations.add(gsd_dumper)
+    sim.operations.schedule()
+    sim.run(1)
+    overlaps = sim.operations.integrator.overlaps
+    assert overlaps == 0
+
+    s = sim.state.snapshot
+    s.particles.position[0] = (0, 0, 0)
+    s.particles.position[1] = (0, 0.85, 0)
+    sim.state.snapshot = s
+    sim.operations.add(mc)
+    # gsd_dumper = hoomd.dump.GSD(filename='/Users/danevans/hoomd/
+    # test_dump_convex_polygon.gsd', trigger=1, overwrite=True)
+    # gsd_logger = hoomd.logger.Logger()
+    # gsd_logger += mc
+    # gsd_dumper.log = gsd_logger
+    # sim.operations.add(gsd_dumper)
+    sim.operations.schedule()
+    sim.run(1)
+    overlaps = sim.operations.integrator.overlaps
+    assert overlaps > 0
+
+
+def test_shape_moves(device, dummy_simulation_check_moves):
+    mc = hoomd.hpmc.integrate.ConvexPolygon(23456)
+    mc.shape['A'] = dict(vertices=[(0, (0.75**0.5) / 2),
+                                   (-0.5, -(0.75**0.5) / 2),
+                                   (0.5, -(0.75**0.5) / 2)])
+    sim = dummy_simulation_check_moves()
```

Comment:
I would suggest several changes to the simulation factories.
1) Remove `dummy_` from the prefix. These are not dummy simulations, but simulations for a specific purpose.
2) 2D shapes (like convex polygon) need particles placed in only in the z=0 plan and the dimensionality of the simulation should be set to 2. You will need to add a `dimensions` argument to the factory function to enable this.
3) Let's generalize the fixtures sufficiently so that they may be used by MD and other simulations. 
  * `dummy_simulation_check_moves` should be `lattice_simulation_factory(particle_types=['A'], dimensions=3, L=20, n=7, a=3)` - where L is the box length, dimension sets the number of dimensions, n sets the number of particles placed on each side of the cube, and a is the spacing between the particles.
  * `dummy_simulation_check_overlap` can actually use the new `lattice_simulation_factory` if you allow `n` to also be a tuple - you would place `n=(2,1)` particles with `a=0.25` to get the behavior you have now.